types/syncmap/syncmap.go

```diff
@@ -0,0 +1,36 @@
+package syncmap
+
+import (
+	"sync"
+)
+
+// SyncMap is a simple, generic thread-safe map implementation.
+type SyncMap[K comparable, V any] struct {
+	mu sync.RWMutex
+	m  map[K]V
+}
+
+func NewSyncMap[K comparable, V any]() *SyncMap[K, V] {
+	return &SyncMap[K, V]{
+		m: make(map[K]V),
+	}
+}
+
+func (s *SyncMap[K, V]) Load(key K) (V, bool) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	val, ok := s.m[key]
+	return val, ok
+}
+
+func (s *SyncMap[K, V]) Store(key K, value V) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.m[key] = value
+}
+
+func (s *SyncMap[K, V]) Items() map[K]V {
```

Comment:
Looks like `Items` is returning the actual map, which could be unsafely modified by a caller. We should return a copy instead to prevent accidental misuse that creates a race condition.
```
// Items returns a copy of the underlying map.
func (s *SyncMap[K, V]) Items() map[K]V {
	s.mu.RLock()
	defer s.mu.RUnlock()
	
	result := make(map[K]V, len(s.m))
	for k, v := range s.m {
		result[k] = v
	}
	return result
}
```